contracts/governance/extensions/GovernorSequentialProposalId.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {Governor} from "../Governor.sol";
+
+abstract contract GovernorSequentialProposalId is Governor {
+    uint256 private _proposalCount;
+    mapping(uint256 proposalHash => uint256 proposalId) private _proposalIds;
+
+    function getProposalId(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        bytes32 descriptionHash
+    ) public view virtual override returns (uint256) {
+        uint256 proposalHash = hashProposal(targets, values, calldatas, descriptionHash);
+
+        uint256 storedProposalId = _proposalIds[proposalHash];
+        return storedProposalId == 0 ? (_proposalCount + 1) : storedProposalId;
+    }
+
+    function _propose(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        string memory description,
+        address proposer
+    ) internal virtual override returns (uint256) {
+        uint256 proposalHash = hashProposal(targets, values, calldatas, keccak256(bytes(description)));
+        _proposalIds[proposalHash] = ++_proposalCount;
+
+        return super._propose(targets, values, calldatas, description, proposer);
+    }
+
+    /**
+     * @dev Internal function to set the sequential proposal ID for the next proposal. This is helpful for transitioning from another governing system.
+     */
+    function _setProposalCount(uint256 proposalCount) internal virtual {
```

Comment:
This function should probably check that the new value is an increase over the old value. That way, calling that by mistake will not result in double numbering
```suggestion
    function _setProposalCount(uint256 proposalCount) internal virtual {
        if (proposalCount < _proposalCount) revert GovernorSequentialProposalIdTooLow();
```

Reply:
@arr00 metionned that setting the `_proposalCount` to `type(uint256).max` would be a way to disable the propose workflow before configuration/migration is done (and `_proposalCount` is reset to a "correct value")

IMO this enabling/disabling should probably be achieved by changing the proposal threshold.

Reply:
added a requirement that the proposal count is strictly increasing